### PR TITLE
fix: compile, env, and other issues

### PR DESCRIPTION
**Pull request by okikio**
- Fix wasmbuild not working due to a version mismatch
- Add Inter.ttf file, so the demo works without needing to manually go download the font file
- Add an env.example file so peeps know what env variables they need to run the demo
- Add a deno task for running the compiled executable with the proper env vars, via `run:standalone`
- Add a `MUSIC_SERVICE` env var for mac os users who only have spotify
- Add `user-read-playback-state` to spotify auth, this is due to spotify requiring playback to specify the device_id for the device you wish to play music off of
- Move a modified version of jxa.ts into the codebase so as to have it use the new `new Deno.Command(...)` API instead of `Deno.run`
- Removed all usage of `localStorage` as the deno compiled executable doesn't support the API

✌🏽

### CODE REVIEW

File: spotify.ts [hunk 1]
```diff
@@ -24,22 +26,23 @@ let client: Client;
...
 }
 
 function getItem(k) {
   try {
-    return localStorage.getItem(k);
+    // return localStorage.getItem(k);
   } catch {
     return null;
   }
 }
```
Comment (littledivy) [inline spotify.ts]: I don't think this is needed. The wrappers here ensure the app continues to work without localStorage.

File: spotify.ts [hunk 2]
```diff
@@ -24,22 +26,23 @@ let client: Client;
...
 
 export async function getSavedTracks() {
   const refreshMeta = getItem("refreshMeta");
+  console.log("refreshMeta", refreshMeta);
```
Comment (littledivy) [inline spotify.ts]: Remove

File: spotify.ts [hunk 3]
```diff
@@ -86,6 +89,9 @@ export async function getSavedTracks() {
   }
 
   const tracks = await client.user.getSavedTracks();
+  console.log({
+    tracks,
+  })
```
Comment (littledivy) [inline spotify.ts]: Remove

File: spotify.ts [hunk 4]
```diff
@@ -98,10 +104,32 @@ export async function getSavedTracks() {
...
+}
+
+export async function getAvailableDevices(): Promise<AvailableDevicesResponse> {
+  return await client!.fetch(`/me/player/devices`);
+}
+
 export async function playTrack({ uri }: { uri: string }) {
-  await client!.fetch(`/me/player/play`, {
+  const { devices = [] } = await getAvailableDevices();
+
```
Comment (littledivy) [inline spotify.ts:123]: Print `No active device found` if devices[] is empty.

File: spotify.ts [hunk 5]
```diff
@@ -98,10 +104,32 @@ export async function getSavedTracks() {
...
+  let x = await client!.fetch(`/me/player/play?device_id=${devices?.[0]?.id}`, {
     method: "PUT",
     headers: { "Content-Type": "application/json" },
     body: { uris: [uri], position_ms: 0 },
   });
+  console.log(x);
```
Comment (littledivy) [inline spotify.ts]: Remove

File: .vscode/settings.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+  "deno.enable": true
+}
```
Comment (littledivy) [inline .vscode/settings.json]: Remove `.vscode`

File: spotify.ts [hunk 7]
```diff
@@ -12,7 +12,9 @@ const scopes = [
   "user-read-private",
   "user-library-read",
   "playlist-read-private",
-  "user-modify-playback-state",
+  // "user-read-currently-playing",
```
Comment (littledivy) [inline spotify.ts]: Remove

File: apple-music.ts [hunk 8]
```diff
@@ -1,4 +1,5 @@
-import { run } from "https://raw.githubusercontent.com/NextFire/jxa/v0.0.4/run/mod.ts";
+// import { run } from "https://raw.githubusercontent.com/NextFire/jxa/v0.0.4/run/mod.ts";
+import { run } from "./jxa.ts";
```
Comment (littledivy) [inline apple-music.ts]: I don't want to vendor dependencies. I think it should be fine to keep using the deprecated API until Deno 2. We can switch to another jxa dep

File: deno.jsonc [hunk 9]
```diff
@@ -5,6 +5,11 @@
   "tasks": {
     "wasmbuild": "deno run -A https://deno.land/x/wasmbuild/main.ts --out ttf/",
     "run": "deno run -A --env --unstable-ffi --unstable-webgpu music-app.tsx",
-    "compile": "deno compile -A --no-check --env --unstable-ffi --unstable-webgpu music-app.tsx --standalone"
+    "compile": "deno compile -A --no-check --env --unstable-ffi --unstable-webgpu music-app.tsx --standalone",
+    
+    // Run `deno task compile` first to generate the standalone executable.
+    // We use `npm:dotenv-cli` to load the environment variables from `.env`,
+    // as deno compiled executables don't support `--env` flag.
+    "run:standalone": "deno run -A npm:dotenv-cli ./music-app"
```
Comment (littledivy) [inline deno.jsonc]: We already use std/dotenv in the compiled app. Wrapping with `npm:dotenv-cli` is not needed.

File: music-app.tsx [hunk 10]
```diff
@@ -10,16 +10,19 @@ import {
   playTrack as playSpotifyTrack,
 } from "./spotify.ts";
 
+import "https://deno.land/std@0.213.0/dotenv/load.ts";
```
Comment (littledivy) [inline music-app.tsx]: We already load once in `spotify.ts`. This can be removed.

**Review by littledivy — COMMENTED** [review]
Thanks for the PR. Left some comments

File: spotify.ts (hunk 1 above)
Comment (okikio) [reply spotify.ts]: The compiler doesn't allow for the localStorage API at all, it crashes if it see it

File: deno.jsonc (hunk 9 above)
Comment (okikio) [reply deno.jsonc]: The compiled application doesn't support it currrently, so this is a quick fix until proper support is added, I can remove it if you feel its not necessary

File: music-app.tsx (hunk 10 above)
Comment (okikio) [reply music-app.tsx]: Oh nice catch

File: spotify.ts (hunk 1 above)
Comment (littledivy) [reply spotify.ts]: Oh, how can I reproduce it?

It works for me from `main`:
```
deno task compile
./music-app
```

Comment (okikio) [reply spotify.ts]: Oh, I can't re-create the issue anymore...hmmm...🤔

Comment (okikio) [reply spotify.ts]: I've uncommented the places localStorage is used

File: apple-music.ts (hunk 8 above)
Comment (okikio) [reply apple-music.ts]: Yep, done

**Comment by okikio** [comment]
Done, I've addressed all comments PTAL

**Review by littledivy — APPROVED** [review]
LGTM, thanks!

**Pull request merged**